modules/match.js

```diff
@@ -37,8 +47,20 @@ function match({ history, routes, location, ...options }, callback) {
     callback(
       error,
       redirectLocation,
-      nextState && { ...nextState, history, router }
+      nextState && {
+        ...nextState,
+        history,
+        router,
+        matchContext: { history, transitionManager, router }
```

Comment:
Well, `router` as well – it's just that in this case, it feels more opaque to delegate everything to the `matchContext` prop; the affordance is that even if the user somehow directly passes in a different `history` or `router` object, that `<Router>` won't silently use those.